packages/expo-sqlite/android/src/main/java/expo/modules/sqlite/SQLiteModule.kt

```diff
@@ -0,0 +1,326 @@
+// Copyright 2015-present 650 Industries. All rights reserved.
+package expo.modules.sqlite
+
+import android.content.Context
+import android.database.Cursor
+import android.database.sqlite.SQLiteDatabase
+import android.database.sqlite.SQLiteStatement
+import org.unimodules.core.ExportedModule
+import org.unimodules.core.Promise
+import org.unimodules.core.interfaces.ExpoMethod
+import java.io.File
+import java.io.IOException
+import java.util.*
+
+class SQLiteModule(private val mContext: Context) : ExportedModule(mContext) {
+  override fun getName(): String {
+    return "ExponentSQLite"
+  }
+
+  @ExpoMethod
+  fun exec(dbName: String, queries: ArrayList<ArrayList<Any>>, readOnly: Boolean, promise: Promise) {
+    try {
+      val numQueries = queries.size
+      val results = arrayOfNulls<SQLitePluginResult>(numQueries)
+      val db = getDatabase(dbName)
+      for (i in 0 until numQueries) {
+        val sqlQuery = queries[i]
+        val sql = sqlQuery[0] as String
+        val bindArgs = convertParamsToStringArray(sqlQuery[1])
+        try {
+          if (isSelect(sql)) {
+            results[i] = doSelectInBackgroundAndPossiblyThrow(sql, bindArgs, db)
+          } else { // update/insert/delete
+            if (readOnly) {
+              results[i] = SQLitePluginResult(EMPTY_ROWS, EMPTY_COLUMNS, 0, 0, ReadOnlyException())
+            } else {
+              results[i] = doUpdateInBackgroundAndPossiblyThrow(sql, bindArgs, db)
+            }
+          }
+        } catch (e: Throwable) {
+          if (DEBUG_MODE) {
+            e.printStackTrace()
+          }
+          results[i] = SQLitePluginResult(EMPTY_ROWS, EMPTY_COLUMNS, 0, 0, e)
+        }
+      }
+      val data = pluginResultsToPrimitiveData(results)
+      promise.resolve(data)
+    } catch (e: Exception) {
+      promise.reject("SQLiteError", e)
+    }
+  }
+
+  @ExpoMethod
+  fun close(dbName: String, promise: Promise) {
+    DATABASES[dbName]!!.close()
+    DATABASES.remove(dbName)
+    promise.resolve(null)
+  }
+
+  // do a update/delete/insert operation
+  private fun doUpdateInBackgroundAndPossiblyThrow(
+    sql: String,
+    bindArgs: Array<String?>?,
+    db: SQLiteDatabase?
+  ): SQLitePluginResult {
+    var statement: SQLiteStatement? = null
```

Comment:
`SQLiteStatement ` should be `autocloseable`. So you can use `use` function - https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/use.html.